koswat/configuration/io/surroundings_wrapper_collection_importer.py

```diff
@@ -1,50 +1,51 @@
 import logging
+from dataclasses import dataclass
 from itertools import groupby
 from pathlib import Path
 
 from koswat.configuration.io.csv.koswat_surroundings_csv_fom import (
     KoswatSurroundingsCsvFom,
-    KoswatSurroundingsWrapperCsvFom,
 )
 from koswat.configuration.io.csv.koswat_surroundings_csv_reader import (
     KoswatSurroundingsCsvReader,
 )
-from koswat.configuration.io.ini.koswat_general_ini_fom import SurroundingsSectionFom
+from koswat.configuration.io.ini.koswat_general_ini_fom import (
+    InfrastructureSectionFom,
+    SurroundingsSectionFom,
+)
 from koswat.configuration.io.shp.koswat_dike_locations_shp_fom import (
     KoswatDikeLocationsShpFom,
 )
 from koswat.configuration.io.shp.koswat_dike_locations_shp_reader import (
     KoswatDikeLocationsListShpReader,
 )
-from koswat.core.io.koswat_importer_protocol import KoswatImporterProtocol
+from koswat.core.protocols.builder_protocol import BuilderProtocol
 from koswat.dike.surroundings.wrapper.surroundings_wrapper import SurroundingsWrapper
 from koswat.dike.surroundings.wrapper.surroundings_wrapper_builder import (
     SurroundingsWrapperBuilder,
 )
 
 
-class KoswatSurroundingsImporter(KoswatImporterProtocol):
+@dataclass
+class SurroundingsWrapperCollectionImporter(BuilderProtocol):
+    """
+    This importer actually behaves as a "builder" as it does not directly
+    import from one source but a combination of many. To avoid bypassing
+    the protocol's signatures it inherits from the `BuilderProtocol` instead.
+    """
+
+    surroundings_section_fom: SurroundingsSectionFom
+    infrastructure_section_fom: InfrastructureSectionFom
+
     traject_loc_shp_file: Path
     selected_locations: list[str]
 
-    def __init__(self) -> None:
-        self.traject_loc_shp_file = None
-        self.selected_locations = []
-
     def _get_dike_locations_shp_fom(self) -> list[KoswatDikeLocationsShpFom]:
         _reader = KoswatDikeLocationsListShpReader()
         _reader.selected_locations = self.selected_locations
         return _reader.read(self.traject_loc_shp_file)
 
-    def import_from(
-        self, surroundings_section: SurroundingsSectionFom
-    ) -> list[SurroundingsWrapper]:
-        _from_path = surroundings_section.surroundings_database_dir
-        if not isinstance(_from_path, Path):
-            raise ValueError("No surroundings csv directory path given.")
-        if not isinstance(self.traject_loc_shp_file, Path):
-            raise ValueError("No traject shp file path given.")
-
```

Comment:
Don't we need these checks anymore?

Reply:
Could be argued. We are now initializing a different way, and the locations are checked later on. If it becomes a problem we can bring again the raises.